File: stack/stack.py
class  Stack:
    def __init__(self, capacity):
        self.capacity = capacity
        self.stack = []
        self.top = -1
    def isFull(self):
        if self.top == self.capacity - 1:
            return True
        else:
            return False
    
    def isEmpty(self):
        if  self.top == -1:
            return True
        else:
            return False
    def push(self, item):
        if self.isFull():
            print("Stack is Full. Elements cant be pushed...")
        else:
            self.stack.append(item)
            self.top += 1
    def pop(self):
        if self.isEmpty():
            print("Stack is Empty. Pop operation cant be performed...")
        else:
            temp = self.stack[self.top]
            del self.stack[self.top]
            self.top -= 1
            return temp
        
    def size(self):
        return self.top + 1
    
    def peek(self):
        if self.isEmpty():
            print("Stack is empty...")
        else:
            return self.stack[self.top]
    
    def printStack(self):
        print("Elements in stack: ")
        for elem in self.stack:
            print(elem)

if __name__ == "__main__":
    stk = Stack(5)
    print(stk.peek())
    print('pop',stk.pop())
    stk.push(-2)
    stk.push(5)
    stk.printStack()
    print(stk.peek())
    stk.pop()
    stk.printStack()
    stk.push(10)
    stk.push(15)
    stk.push(20)
    stk.push(25)
    stk.printStack()
    print(stk.peek())
    stk.push(30)





    
    
    

File: stack/stack.java
public class Stack{
    private int capacity, top;
    private int stack[];
    Stack(int capacity){
        this.capacity = capacity;
        this.top = -1;
        this.stack = new int[capacity];
    }
    public Boolean isEmpty(){
        return this.top == -1;
    }
    public Boolean isFull() {
        return this.top + 1 == this.capacity;
    }
    public void push(int item) {
        if (this.isFull()) {
            System.out.println("The stack is full. Push operation cant be performed...");
            System.exit(1);
        } else {
            this.stack[++this.top] = item;
        }
    }
    public int pop(){
        if (this.isEmpty()) {
            System.out.println("The stack is Empty. Pop operation cant be performed...");
            System.exit(1);
        }
        return this.stack[this.top--];

    }
    public int size(){
        return this.top +1;
    }
    public int peek(){
        if (this.isEmpty()){
            System.out.println("Stack is empty nothing to peek");
            System.exit(1);
        }
        return this.stack[this.top];
    }
    public void printStack(){
        for (int item : this.stack){
            System.out.println(item);
        }
    }
    public static void main(String args[]){
        Stack stk = new Stack(5);
        System.out.println(stk.peek());
        System.out.println("pop "+ stk.pop());
        stk.push(-2);
        stk.push(5);
        stk.printStack();
        System.out.println(stk.peek());
        stk.pop();
        stk.printStack();
        stk.push(10);
        stk.push(15);
        stk.push(20);
        stk.push(25);
        stk.printStack();
        System.out.println(stk.peek());
        stk.push(30);
        System.out.println(stk.size());

    }
}
